core/src/eav/memory/mod.rs

```diff
@@ -0,0 +1,212 @@
+use eav::{Attribute, Entity, EntityAttributeValue, EntityAttributeValueStorage, Value};
+use error::HolochainError;
+use std::collections::HashSet;
+
+pub struct MemoryStorageEav {
+    eavs: HashSet<EntityAttributeValue>,
+}
+
+impl MemoryStorageEav {
+    pub fn new() -> MemoryStorageEav {
+        MemoryStorageEav {
+            eavs: HashSet::new(),
+        }
+    }
+}
+
+impl EntityAttributeValueStorage for MemoryStorageEav {
+    fn add_eav(&mut self, eav: &EntityAttributeValue) -> Result<(), HolochainError> {
+        self.eavs.insert(eav.clone());
+        Ok(())
+    }
+    fn fetch_eav(
+        &self,
+        entity: Option<Entity>,
+        attribute: Option<Attribute>,
+        value: Option<Value>,
+    ) -> Result<HashSet<EntityAttributeValue>, HolochainError> {
+        Ok(self
+            .eavs
+            .iter()
+            .cloned()
```

Comment:
Is this needed?

Reply:
Right, but doesn't the collect at the end create a new HashSet since it's not into_iter()?

Reply:
Yes absolutely but not of type EntityAttributeValue they would be of &EntityAttributeValue :(